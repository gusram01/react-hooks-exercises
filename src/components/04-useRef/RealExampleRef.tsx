import React, { useState } from 'react';
import './focus.css';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  return (
    <>
      <h1>Real Example Ref</h1>
      <hr />
      {
        show && <MultipleCustomHooks />
      }

      <button
        className="btn btn-outline-primary mt-5"
        onClick={() => { setShow(!show) }}
      >
        Show / Hide
      </button>
    </>
  );
};

export default RealExampleRef;