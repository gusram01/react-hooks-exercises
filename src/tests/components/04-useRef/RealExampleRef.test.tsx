import React from 'react';
import { shallow } from 'enzyme';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';

describe('RealExampleRef tests', () => {
  const wrapper = shallow(<RealExampleRef />)

  test('RealExampleRef should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render MultipleCustomHooks correctly', () => {
    wrapper.find('.btn').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});
