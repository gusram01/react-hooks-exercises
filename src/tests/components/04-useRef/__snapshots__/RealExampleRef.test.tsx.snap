// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef tests RealExampleRef should render correctly 1`] = `
<Fragment>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</Fragment>
`;
